### ISSUE TITLE
regression with UnknownInitialization

### ISSUE THREAD

**Issue by cushon** [typetools/checker-framework#4381]
This diagnostic is reported in 3.11.0 but not 3.10.0, is this a bug? Since `m` is a local variable I don't think there's anything here that couldn't be initialized, and when I was trying to minimize it, it seems like an interaction with the `try`/`catch`.

```java
import java.io.IOException;
import java.security.GeneralSecurityException;

abstract class T {

  public void t() {
    int m = 0;
    try {
      f();
    } catch (IllegalArgumentException | IOException e) {
    } catch (GeneralSecurityException e) {
      g(m);
    }
  }

  abstract void g(int x);

  abstract void f() throws IllegalArgumentException, IOException, GeneralSecurityException;
}
```

```
$ ./checker-framework-3.11.0/checker/bin/javac -processor Nullness T.java
...
T.java:12: error: [argument.type.incompatible] incompatible argument for parameter x of g.
      g(m);
        ^
  found   : @UnknownInitialization @NonNull int
  required: @Initialized @NonNull int
1 error
```

**Comment by mernst** [comment]
The faulty commit seems to be 6d83a13d22114e4f03c75f3fa6c12f1dbc56c493.
